docs/toolkits/openstack-operations-maintenance.md

```diff
@@ -1,47 +1,181 @@
+
+
+
+### Maintenance
+
+---
+
+#### NTP
+	- What NTP Errors look like
+		The biggest offender here is going to be any clock drift errors in the logs and odd behavior by computes on some isolated VMs.
+	- How to fix
+		Usually this is either two issues.  First, the time daemon running on your node may simply need a restart.  The second is that it could simply be a misconfiguration on said time daemon.
+
+#### Rabbit MQ
+
+	- Checks:
+		To check the status of your RabbitMQ cluster, run the following on one of the rabbitmq nodes:
+``` bash
+rabbitmqctl cluster_status
+```
+Some things to look for here, are any errors or alarms under the [Alarms] section.
+
+	- What rabbitMQ errors look like
+		rabbitMQ errors can be tough to diagnose.  All the openstack services send messages through rabbitMQ, so there is a broad range of errors that an administrator can come across while trying to determine the cause.
+	- How to fix
+		The simplest fix is usually going to be a dead rabbitMQ service on one of the nodes.  Usually a restart can fix these.
+
+#### Mysql
+
+	- Checks for a single node database:
 SSH to the MYSQL node and log into the database with:
 `mysql`
 Also, you can check the logs to make sure there are no discrepancies.
 
-Checks for a multiple node cluster:
+	- Checks for a multiple node cluster:
 SSH to the MYSQL node and log into the database with:
 `mysql`
 and run
 ```
 SHOW GLOBAL STATUS LIKE 'wsrep_%';
 ```
-
 And verify the cluster count matches the number of nodes in your database cluster.
 
-Nova
+	- What MYSQL errors look like
+		If MYSQL is causing issues, you'll usually get a 503 error when attempting to run commands from the CLI.
+	- How to fix
+		This could be as simple as a dead mysql daemon on one of your nodes.  You can diagnose this by running through the checks above.  
+	- Database Discrepancies
+		Throughout the life of your cloud, you may come into issues where the database and openstack reflect different information.  This could be a VM that shows a host on one machine but it's actually on another, a volume that is stuck in a `reserved` state but won't change back to active/available, etc.  To fix this, we need to log into the MYSQL cluster by running `mysql` and updating the discrepancy with `update ITEM set VALUE=DESIRED_VALUE  where id='UUID';`.  You can verify which specific item(s) you'll adjust by adjusting the formula to, `show ITEM where id='UUID';`.
 
+#### Nova
+
+	- Checks:
 To check the nova status, source the Admin openrc file or app credential and run:
 ```
 openstack compute service list
 ```
 
 This will list all nova services and show if any are down or not running.  A "XXX" status simply means that service has missed a heartbeat message in rabbitmq to the control node.  A "down" status means the service has missed enough to be considered off or problematic.
+	- What Nova errors look like
+		Nova errors will generally be the case where you are getting "No valid hosts available" when their should be.
```

Comment:
```suggestion
	- What Nova errors look like
		Nova errors will generally be the case where you are getting "No valid hosts available", when there should be hosts available.
```

Also, if you run `mkdocs` you'll see it fails to format this whole section, I believe you can fix by adding a blank line before the first bullet (and de-denting bullets per above).

Reply:
I can drop in a link to the OpenStack app credential page.